docs/CODING_STYLE.rst

```diff
@@ -67,32 +67,219 @@ Coding guide
+     *                        (NULL for no/invisible mouse cursor)
+     *
+     * @note  Include any subtle points an API user or maintainer would need to know.
+     */
+    void lv_other_example(lv_display_t * disp, lv_image_dsc_t const * mouse_img);
 
-Short "code summaries" of a few lines are accepted. E.g.
-``/*Calculate the new coordinates*/``
+When documenting a code member that is *before* the comment, such as a struct member,
+use ``/**<`` like this:
 
-In comments use \` \` when referring to a variable. E.g.
-:literal:`/\*Update the value of \`x_act`*/`
+.. code-block:: c
+
+    /**
+     * Brief description of struct
+     */
+    typedef struct _lv_example_type_t {
+        char      *text;    /**< Brief description of this member */
+        uint16_t   length;  /**< Brief description of this member */
+    } lv_example_type_t;
+
+| When commenting code, use block comments like this ``/* Description */``,
+| not end-of-line comments like this ``// Description``.
+
+Include a space after the ``/*`` or ``/**<`` and before the ``*/`` for added readability.
+
+Write readable code to avoid descriptive comments like:  ``x++; /* Add 1 to x */``.
+
+The code should show clearly what you are doing.
+
+You should write **why** you did it:  ``x++;  /* Point to closing '\0' of string */``
+
+Short "code summaries" of a few lines are accepted: ``/* Calculate new coordinates */``
+
+In comments use \`...\` when referring to a variable: ``/* Update value of `x_act` */``
+
+When adding or modifying comments, priorities are (in order of importance):
+
+    1.  clarity (the ease with which other programmers can understand your intention),
+    2.  readability (the ease with which other programmers can read your comments),
+    3.  brevity (the quality of using few words when speaking or writing).
+
+Blank lines within comments are desirable when they improve clarity and readability.
+
+Remember, when you are writing source code, you are not just teaching the computer
+what to do, but also teaching other programmers how it works, not only users of the
+API, but also future maintainers of your source code.  Comments add information
+about what you were thinking when the code was written, and **why** you did things
+that way -- information that cannot be conveyed by the source code alone.
+
+
+Doxygen Comment Specifics
+~~~~~~~~~~~~~~~~~~~~~~~~~
+Doxygen is the first program in a data-flow chain that generates the online LVGL
+API documentation from the files in the LVGL repository.  Doxygen detects comments
+it should pay attention to by leading ``/**``.  It ignores comments that do not
+have exactly two ``*``.
+
+    | ``/** Description of code member AFTER this comment. */``
+    | ``code_member_here {...}`` (e.g. function, struct, etc.)
+
+    ``variable_or_struct_member;  /**< Description of code member BEFORE this comment. */``
+
+Add 2 spaces after Doxygen commands (they start with '@') for improved readability.
```

Comment:
Now we add only 1 space, e.g. `@param value`

Reply:
> Now we add only 1 space, e.g. `@param value`

I know, and in the code, it is really difficult to read in some cases!

Imagine you know _nothing_ about this function and then compare this:
```
/**
 * Set how to place (where to align) the items and tracks
 * @param obj pointer to an object. The parent must have flex layout else nothing will happen.
 * @param main_place where to place the items on main axis (in their track). Any value of `lv_flex_align_t`.
 * @param cross_place where to place the item in their track on the cross axis. `LV_FLEX_ALIGN_START/END/CENTER`
 * @param track_cross_place where to place the tracks in the cross direction. Any value of `lv_flex_align_t`.
 */
void lv_obj_set_flex_align(lv_obj_t * obj, lv_flex_align_t main_place, lv_flex_align_t cross_place,
                           lv_flex_align_t track_cross_place);
```
with this
```
/**
 * Set how to place (where to align) the items and tracks
 * @param  obj                pointer to an object. The parent must have flex layout else nothing will happen.
 * @param  main_place         where to place the items on main axis (in their track). Any value of `lv_flex_align_t`.
 * @param  cross_place where  to place the item in their track on the cross axis. `LV_FLEX_ALIGN_START/END/CENTER`
 * @param  track_cross_place  where to place the tracks in the cross direction. Any value of `lv_flex_align_t`.
 */
void lv_obj_set_flex_align(lv_obj_t * obj, lv_flex_align_t main_place, lv_flex_align_t cross_place,
                           lv_flex_align_t track_cross_place);
```

I think we should promote the 2nd way (rather than promoting what is in the code -- which is part of why it is difficult, sometimes painful [no I am not exaggerating] to learn LVGL).

Of course, if you're thinking about making the documentation _align_ with the code for v9.2, I can understand that, but I am thinking about the future.

Thoughts?

Reply:
Wow, it really looks better. I vote for `example 3`. `Example 4` might be ok too, but I'm not sure so let's stay on the safe side and go for `example 3`. Once it's merged we add the missing spaces with a script.

Reply:
> Wow, it really looks better. I vote for `example 3`. `Example 4` might be ok too, but I'm not sure so let's stay on the safe side and go for `example 3`. Once it's merged, in a new PR we can add the missing spaces with a script.

Agreed!  It seems we think very similarly--harmony first, then advance the library by increments.